### PR TITLE
[Kestrel] Add missing XML docs to public APIs

### PR DESCRIPTION
**Pull request by javiercn**

### CODE REVIEW

**Review by Tratcher — APPROVED** [review]

File: src/Servers/Connections.Abstractions/src/DefaultConnectionContext.cs [hunk 1]
```diff
@@ -23,16 +26,20 @@ public class DefaultConnectionContext : ConnectionContext,
...
         {
         }
 
         /// <summary>
-        /// Creates the DefaultConnectionContext without Pipes to avoid upfront allocations.
+        /// Creates the <see cref="DefaultConnectionContext"/> without Pipes to avoid upfront allocations.
         /// The caller is expected to set the <see cref="Transport"/> and <see cref="Application"/> pipes manually.
         /// </summary>
-        /// <param name="id"></param>
+        /// <param name="id">The connection id.</param>
```
Comment (halter73) [inline src/Servers/Connections.Abstractions/src/DefaultConnectionContext.cs:42]:
```suggestion
        /// <param name="id">The <see cref="ConnectionId"/>.</param>
```

File: src/Servers/Connections.Abstractions/src/DefaultConnectionContext.cs [hunk 2]
```diff
@@ -48,34 +55,54 @@ public DefaultConnectionContext(string id)
             ConnectionClosed = _connectionClosedTokenSource.Token;
         }
 
+
+        /// <summary>
+        /// Creates the DefaultConnectionContext with the given <paramref name="transport"/> and <paramref name="application"/> pipes.
+        /// </summary>
+        /// <param name="id">The connection id.</param>
```
Comment (halter73) [inline src/Servers/Connections.Abstractions/src/DefaultConnectionContext.cs:62]:
```suggestion
        /// <param name="id">The <see cref="ConnectionId"/>.</param>
```

File: src/Servers/Connections.Abstractions/src/DefaultConnectionContext.cs [hunk 3]
```diff
@@ -48,34 +55,54 @@ public DefaultConnectionContext(string id)
...
+        /// <param name="transport">The transport pipe.</param>
```
Comment (halter73) [inline src/Servers/Connections.Abstractions/src/DefaultConnectionContext.cs:63]:
```suggestion
        /// <param name="transport">The <see cref="Transport"/>.</param>
```

File: src/Servers/Connections.Abstractions/src/DefaultConnectionContext.cs [hunk 4]
```diff
@@ -48,34 +55,54 @@ public DefaultConnectionContext(string id)
...
+        /// <param name="application">The application pipe.</param>
```
Comment (halter73) [inline src/Servers/Connections.Abstractions/src/DefaultConnectionContext.cs:64]:
```suggestion
        /// <param name="transport">The <see cref="Application"/>.</param>
```

File: src/Servers/Connections.Abstractions/src/Features/IConnectionEndpointFeature.cs [hunk 5]
```diff
@@ -5,9 +5,19 @@
 
 namespace Microsoft.AspNetCore.Connections.Features
 {
+    /// <summary>
+    /// A feature that represents a connection endpoint.
```
Comment (halter73) [inline src/Servers/Connections.Abstractions/src/Features/IConnectionEndpointFeature.cs:9]:
```suggestion
    /// A feature that represents a connection endpoints.
```

File: src/Servers/Connections.Abstractions/src/Exceptions/ConnectionAbortedException.cs [hunk 6]
```diff
@@ -1,19 +1,34 @@
-﻿using System;
+using System;
 
 namespace Microsoft.AspNetCore.Connections
 {
+    /// <summary>
+    /// An exception that is thrown when a connection is aborted by the client.
```
Comment (halter73) [inline src/Servers/Connections.Abstractions/src/Exceptions/ConnectionAbortedException.cs:6]:
```suggestion
    /// An exception that is thrown when a connection is aborted by the server.
```

File: src/Servers/Kestrel/Transport.Sockets/src/SocketTransportOptions.cs [hunk 7]
```diff
@@ -40,8 +43,14 @@ public class SocketTransportOptions
         /// </remarks>
         public int Backlog { get; set; } = 512;
 
+        /// <summary>
+        /// Gets or sets the maximum read buffer size.
```
Comment (halter73) [inline src/Servers/Kestrel/Transport.Sockets/src/SocketTransportOptions.cs:47]:
```suggestion
        /// Gets or sets the maximum unconsumed incoming bytes the transport will buffer.
```

File: src/Servers/Kestrel/Transport.Sockets/src/SocketTransportOptions.cs [hunk 8]
```diff
@@ -40,8 +43,14 @@ public class SocketTransportOptions
...
+        /// </summary>
         public long? MaxReadBufferSize { get; set; } = 1024 * 1024;
 
+        /// <summary>
+        /// Gets or sets the maximum write buffer size.
```
Comment (halter73) [inline src/Servers/Kestrel/Transport.Sockets/src/SocketTransportOptions.cs:52]:
```suggestion
        /// Gets or sets the maximum outgoing bytes the transport will buffer before applying write backpressure.
```

**Review by halter73 — APPROVED** [review]

File: src/Servers/Connections.Abstractions/src/DefaultConnectionContext.cs [hunk 9]
```diff
@@ -48,34 +55,54 @@ public DefaultConnectionContext(string id)
             ConnectionClosed = _connectionClosedTokenSource.Token;
         }
 
+
+        /// <summary>
+        /// Creates the DefaultConnectionContext with the given <paramref name="transport"/> and <paramref name="application"/> pipes.
+        /// </summary>
+        /// <param name="id">The <see cref="ConnectionId"/>.</param>
+        /// <param name="transport">The <see cref="Transport"/>.</param>
+        /// <param name="transport">The <see cref="Application"/>.</param>
```
Comment (javiercn) [inline src/Servers/Connections.Abstractions/src/DefaultConnectionContext.cs:64]:
```suggestion
        /// <param name="application">The <see cref="Application"/>.</param>
```

Comment (halter73) [reply src/Servers/Connections.Abstractions/src/DefaultConnectionContext.cs]: My bad!

**Pull request merged**